src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -14,15 +18,31 @@ public class PoAHeaderSignatureRule : HeaderValidationConsensusRule
 
         private SlotsManager slotsManager;
 
+        private uint maxReorg;
+
+        private bool votingEnabled;
+
+        private VotingManager votingManager;
+
+        private FederationManager federationManager;
+
+        private IChainState chainState;
+
         /// <inheritdoc />
         public override void Initialize()
         {
             base.Initialize();
 
-            var engine = this.Parent as PoAConsensusRuleEngine;
+            PoAConsensusRuleEngine engine = this.Parent as PoAConsensusRuleEngine;
 
             this.slotsManager = engine.SlotsManager;
-            this.validator = engine.poaHeaderValidator;
+            this.validator = engine.PoaHeaderValidator;
+            this.votingManager = engine.VotingManager;
+            this.federationManager = engine.FederationManager;
+            this.chainState = engine.ChainState;
+
+            this.maxReorg = this.Parent.Network.Consensus.MaxReorgLength;
+            this.votingEnabled = ((PoAConsensusOptions) this.Parent.Network.Consensus.Options).VotingEnabled;
```

Comment:
Can this be enabled after the network has started?

Reply:
Yes, but all fed members will have to do that, otherwise there will be a fork after the first applied poll.